src/adapters/Cornerstone/Segmentation_4X.js

```diff
@@ -629,6 +629,43 @@ function checkSEGsOverlapping(
     var groupsLen = PerFrameFunctionalGroupsSequence.length;
     var numberOfSegs = multiframe.SegmentSequence.length;
     var numberOfFrames = imageIds.length;
+    // NOTE:
+    // Assumption1 : numberOfSegs * numberOfFrames = groupsLen,
+    // i.e. for each frame we have a N PerFrameFunctionalGroupsSequence, where N is the numberOfSegs.
+    // Assumption2 : the order of the group sequence is = numberOfFrames of segmentation 1 +  numberOfFrames of segmentation 2 + ... + numberOfFrames of segmentation numberOfSegs
+    // -------------------
+    // We could check the ImagePositionPatient and working in 3D coordinates (instead of indexes) and remove the assumptions,
+    // but this would greatly increase the computation time (i.e. we would have to sort the data before running checkSEGsOverlapping).
+
+    if (numberOfSegs * numberOfFrames !== groupsLen) {
+        console.warn(
+            "checkSEGsOverlapping: it not possible performing the overlapping segments check since PerFrameFunctionalGroupsSequence has missing data."
+        );
+        return false;
+    }
+
+    const refSegFrame0 =
+        PerFrameFunctionalGroupsSequence[0] &&
+        PerFrameFunctionalGroupsSequence[0].SegmentIdentificationSequence
+            ? PerFrameFunctionalGroupsSequence[0].SegmentIdentificationSequence
+                  .ReferencedSegmentNumber
+            : undefined;
+    const refSegFrame1 =
+        PerFrameFunctionalGroupsSequence[1] &&
```

Comment:
In either case, we should avoid duplication of this ternary and move it into a new function.